projects/copra-finance/index.js

```diff
@@ -11,15 +11,151 @@ async function tvl(_, _b, _cb, { api }) {
     eventAbi: 'event CreditAccountDeployed (address indexed creditAccount)',
     fromBlock,
   })
-  const owners = logs.map((i) => i.creditAccount)
+  const creditAccounts = logs.map((i) => i.creditAccount)
 
-  return api.sumTokens({ owners, tokens });
+  const tvls = await Promise.all(creditAccounts.map(async (creditAccountAddr) => {
+    const status = await getStatus(creditAccountAddr, api)
+    if (status == 0 || status == 1) {
+      const assetAddr = await getCreditAccountAsset(creditAccountAddr, api)
+      const borrowerFundedAmount = await getBorrowerFundedAmount(creditAccountAddr, api);
```

Comment:
this is including borrowed funds as tvl?

Reply:
Not quite.  Our protocol works by having the borrower address deposit a certain amount into the contract first as a security deposit.  `getBorrowerFundedAmount` returns the deposited amount.

Reply:
@cds95 this security deposit is also kept in the same credit account or some other contract? because I tried checking balance in credit account, still getting only 200k

Reply:
So some of the assets in the credit account are deployed to borrower protocols in exchange for LP tokens, which is why we can’t simply read the contract’s asset balance. We are working on a more accurate way to value the LP tokens but we think that converting the amounts deposited into the USD value gives a good estimate for now.

Reply:
can you share an example? we have price support for a lot of LP tokens, also have helper functions that can get underlying token value. 

Thing is, I would like to show both the LP value, and ideally show the borrowed amount separately like we do for aave or compound, as some users like to see how much is borrowed (like compared to ghost lending apps which has a lot of tvl but no borrowing/usage)

also, is there some log event to track the fees paid? we can track it in our fees/revenue metric

Reply:
Sure so currently assets are deposited from the Credit Account to Steadefi, Teahouse and Orange Finance in exchange for LP tokens. 

Can you explain what you mean by displaying the borrowed amount? Do you mean the amount that we have deposited into other protocols?

We currently have not turned on fees for our protocol yet.

Currently we feel that summing up the deposited amounts gives a good estimate of our TVL. If anything it underestimates our TVL as it doesn’t take into account anything our LP tokens earn by depositing into the external protocols.

Reply:
To add to the above, we are applying for LTIP program on Arbitrum and are currently on the evaluation window which include TVL confirmation via DeFiLlama. Hence, we would appreciate if the current solution can be implemented first to show a much more accurate TVL than what is being shown (300k instead of 200k). We are happy to commit working on the ideal solution that you suggested right after. Thanks for understanding.

Reply:
hi @cds95 thanks for patiently explaining to the idiot who mistook a leverage farming protocol to be a lending protocol. I have merged the changes in this PR with minor refactor